File: backend/app/models/help_requests.py
from pydantic import BaseModel, Field

class CreateHelpRequest(BaseModel):
    customer_id: str = Field(..., min_length=1)
    question: str = Field(..., min_length=3)

class HelpRequestOut(BaseModel):
    id: str
    customer_id: str
    question: str
    status: str
    supervisor_answer: str | None = None
    seen_by_supervisor: bool = False

class ResolveHelpRequest(BaseModel):
    answer: str = Field(..., min_length=1)
    resolver: str = Field(..., min_length=1)

File: frontdesk/supervisor-ui/src/types/helpRequests.ts
export type HelpRequestStatus = "pending" | "resolved" | "unresolved";

export type HelpRequest = {
  id: string;
  question: string;
  customer_id: string;
  status: HelpRequestStatus;
  created_at: string;
  updated_at: string;
  // Optional backend flags we may show later:
  ai_followup_sent?: boolean;
  supervisor_answer?: string | null;
  seen_by_supervisor?: boolean;
};
export type ResolvePayload = {
  answer: string;
  resolver: string; // e.g. "supervisor-1"
};
